h/buildext.py

```diff
@@ -219,80 +236,78 @@ def build_chrome(args):
     else:
-        build_extension_common(env)
+        build_extension_common(webassets_env, base_url)
 
     # Render the manifest.
     with codecs.open('build/chrome/manifest.json', 'w', 'utf-8') as fp:
-        data = chrome_manifest(env['request'])
+        script_url = urlparse.urlparse(webassets_env.url)
+        script_host_url = '{}://{}'.format(script_url.scheme,
+                                           script_url.netloc)
+        data = chrome_manifest(script_host_url)
         fp.write(data)
 
     # Write build settings to a JSON file
     with codecs.open('build/chrome/settings-data.js', 'w', 'utf-8') as fp:
+        settings = settings_dict(base_url, api_url, sentry_dsn)
         fp.write('window.EXTENSION_CONFIG = ' + json.dumps(settings))
 
 
 parser = argparse.ArgumentParser('hypothesis-buildext')
-
-parser.add_argument('config_uri', help='paster configuration URI')
-
 parser.add_argument('--debug',
                     action='store_true',
                     default=False,
                     help='create source maps to enable debugging in browser')
-
-subparsers = parser.add_subparsers(title='browser', dest='browser')
-subparsers.required = True
-
-parser_chrome = subparsers.add_parser(
-    'chrome',
-    help="build the Google Chrome extension",
-    description=textwrap.dedent(build_chrome.__doc__),
-    formatter_class=argparse.RawDescriptionHelpFormatter)
-parser_chrome.add_argument('--base',
-                           help='Base URL',
-                           default='http://localhost:5000',
-                           metavar='URL')
-parser_chrome.add_argument('--assets',
-                           help='A path (relative to base) or URL from which '
-                           'to load the static assets',
-                           default=None,
-                           metavar='PATH/URL')
+parser.add_argument('--sentry-dsn',
+                    default='',
+                    help='Specify the Sentry DSN for crash reporting',
+                    metavar='SENTRY_DSN')
+parser.add_argument(
+    '--service',
+    help="""The URL of the Hypothesis service which the extension
+                            should connect to""",
+    default='http://localhost:5000',
+    dest='service_url',
+    metavar='URL')
+parser.add_argument('--assets',
+                    help='A path (relative to base) or URL from which '
+                    'to load the static assets',
+                    default=None,
+                    metavar='PATH/URL')
+parser.add_argument('browser',
+                    help='Specifies the browser to build an extension for',
+                    choices=['chrome'])
 
 BROWSERS = {'chrome': build_chrome}
 
 
 def main():
-    # Set a flag in the environment that other code can use to detect if it's
-    # running in a script rather than a full web application. See also
-    # h/script.py.
-    #
-    # FIXME: This is a nasty hack and should go when we no longer need to spin
```

Comment:
It's still needed in the other places, I'm afraid. It's stopping app spin-up things happening when they're not meant to.